src/utils/include/utils/quaternion.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef SRC_UTILS_INCLUDE_UTILS_QUATERNION_HPP
+#define SRC_UTILS_INCLUDE_UTILS_QUATERNION_HPP
+
+#include <boost/qvm/deduce_scalar.hpp>
+#include <boost/qvm/quat_operations.hpp>
+#include <boost/qvm/quat_traits.hpp>
+
+#include "utils/Vector.hpp"
+
+namespace Utils {
+template <typename T> class Quaternion : private Vector<T, 4> {
```

Comment:
Couldn't you just use the quaternion class from qvm?

Reply:
you mean this?

``` c++
namespace boost { namespace qvm {

    template <class T>
    struct quat {

      T a[4];

      template <class R>
      operator R() const {
        R r;
        assign(r,*this);
        return r;
      }

    };
}
}

Reply:
Not enough `enable_if`?